lib/ruby_smb/signing.rb

```diff
@@ -51,6 +51,8 @@ def smb2_sign(packet)
     # @param [String] session_key The key to use for signing.
     # @return [RubySMB::GenericPacket] the signed packet
     def self.smb2_sign(packet, session_key)
+      return packet if session_key.nil? || session_key == ''
```

Comment:
For feature parity I added this guard clause to ensure the tests continue to pass with with OpenSSL3; but I think the tests are wrong:

```
From: /mnt/hgfs/ruby_smb/spec/lib/ruby_smb/client_spec.rb:2126 :

    2121: 
    2122:       context 'when there is no session_key' do
    2123:         it 'returns the packet exactly as it was given' do
    2124:           smb2_client.session_key = ''
    2125:           smb2_client.signing_required = true
 => 2126:           expect(smb2_client.smb2_sign(request1)).to eq request1
    2127:         end
    2128:       end
    2129:     end
```

Reply:
Context on the failure:

The failing line was here:

https://github.com/rapid7/ruby_smb/blob/master/lib/ruby_smb/signing.rb#L56

This code on OpenSSL 1.1.1:

```
2.7.2 :004 > require 'openssl'; require 'ostruct'; session_key = ''; packet = OpenStruct.new(to_binary_s: 'hello world'); OpenSSL::HMAC.digest(OpenSSL::Digest.new('SHA256'), session_key, packet.to_binary_s)
 => "\xC2\xEAcL\x99?\x05\x04\x82\xB4\xE6$2$\b\x7F|#\xBD\xD3\xC0z\xB1\xA4^\x9A!\xC6/\xAD\x99N" 
```

This code on OpenSSL 3:

```
irb(main):013:0> require 'openssl'; require 'ostruct'; session_key = ''; packet
 = OpenStruct.new(to_binary_s: 'hello world'); OpenSSL::HMAC.digest(OpenSSL::Digest.new('SHA256'), session_key, packet.to_binary_s)
/usr/lib/ruby/3.0.0/openssl/hmac.rb:36:in `initialize': EVP_PKEY_new_mac_key: malloc failure (OpenSSL::HMACError)
	from /usr/lib/ruby/3.0.0/openssl/hmac.rb:36:in `new'
	from /usr/lib/ruby/3.0.0/openssl/hmac.rb:36:in `digest'
	from (irb):13:in `<main>'
	from /var/lib/gems/3.0.0/gems/irb-1.3.6/exe/irb:11:in `<top (required)>'
	from /usr/local/bin/irb:25:in `load'
	from /usr/local/bin/irb:25:in `<main>'
```

We could default the session key to the right length if that makes more sense, instead of using this guard clause?

```
irb(main):017:0> require 'openssl'; require 'ostruct'; session_key = "\x00" * 1
6; packet = OpenStruct.new(to_binary_s: 'hello world'); OpenSSL::HMAC.digest(OpenSSL::Digest.new('SHA256'), session_key, packet.to_binary_s)
=> "\xC2\xEAcL\x99?\x05\x04\x82\xB4\xE6$2$\b\x7F|#\xBD\xD3\xC0z\xB1\xA4^\x9A!\xC6/\xAD\x99N"
```

Reply:
I've gone ahead and used the `"\x00" * 16` session key instead